src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -141,6 +141,42 @@ export const mapTokenUi = ({
 	})
 });
 
+/** Function to sum the balances of two tokens.
+ *
+ * If the decimals of the tokens are the same, the balances are added together.
+ * If the decimals are different, the function returns null.
+ * If one of the balances is nullish, the function returns the other balance.
+ * If both balances are nullish, the function prioritize the first token
+ * NOTE: the function assumes that the two tokens are always 1:1 twins, for example BTC and ckBTC, or ETH and SepoliaETH
+ *
+ * @param token1
+ * @param token2
+ * @returns The sum of the balances or nullish value.
+ */
+export const sumTokenBalances = ([token1, token2]: TokenUi[]): TokenUi['balance'] =>
+	nonNullish(token1.balance) && nonNullish(token2.balance)
+		? token1.decimals === token2.decimals
+			? token1.balance.add(token2.balance)
+			: null
+		: isNullish(token2.balance)
+			? token1.balance
+			: token2.balance;
+
+/** Function to sum the USD balances of two tokens.
+ *
+ * If one of the balances is nullish, the function returns the other balance.
```

Comment:
Does it really matter here? It's all dollars, shouldn't the array just be reduced to sum all usd balance?

Reply:
ah! make sense what about this?

```typescript
export const sumTokenUsdBalances = ([token1, token2]: TokenUi[]): TokenUi['usdBalance'] =>
	nonNullish(token1.usdBalance) || nonNullish(token2.usdBalance)
		? (token1.usdBalance ?? 0) + (token2.usdBalance ?? 0)
		: undefined;
```
		
because `usdBalance` can be either `number` or `undefined`

Reply:
i would prefer to leave `undefined` if both are `undefined`

Reply:
For consistency with `sumTokenBalances`, shouldn't `sumTokenUsdBalances` returns `null` then if both tokens are `undefined`?

Or should the function always returns a zero or value, hence suggesting a reduce and sum?

Reply:
Good question...

The only issue I see is to keep it consistent with definition of `TokenUi.usdBalance` that can be either `number` or `undefined`, but not `null`.

When it is `undefined`, we show it differently in the page `"$-"`, so I thought to keep that same behavior: if both `usdBalance` are `undefined`, the sum is `undefined` and it will showed in the same way `"$-"`

The alternative could be to show `zero`, as you suggested, but I am not sure it could be `null`... WDYT?

Reply:
Makes sense, thanks for the explanation.